### PR TITLE
Dev mode: enable hot reloading without building packages

### PR DESCRIPTION
**Pull request by iceanddust**
### What does this PR do?
- It makes use of the Vite feature to treat monorepo linked packages as source code, allowing hot reloading in dev mode without needing to build packages on each change. This translates into a smoother dev experience and faster development as the app reloading is almost immediate on each change.
- Add a root `tsconfig.json` file and make each package and app's tsconfig file inherit from the root one.
- Switch the Vite compiler plugin from Babel to SWC. This is necessary for Lingui to work on the linked packages (i.e. the Widgets package), and it has the added benefit of being faster than Babel on app bundling and compiling.
- Update the size of some icons: there were some icons that didn't seem to be working properly based on the Chakra Icon props, after switching from Babel to SWC these properties were applied according to Chakra docs, causing the icons to break in size. Hence, they needed to be resized after the update.

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -3,14 +3,8 @@
   "version": "1.0.0",
   "description": "",
   "type": "module",
-  "module": "./dist/makerdao-dux-components.js",
```
Comment (generatorpoint) [inline packages/components/package.json:6]: crazy, I've been seeing references to this in the logs but I had no idea where they were coming from

**Review by generatorpoint — APPROVED** [review]
this is awesome, hot reloading is crazy fast now.

**Review by inclinedmind — APPROVED** [review]
Awesome, great job finding a solution for this

**Review by bitsforpeace — APPROVED** [review]
In testing, I'm noticing an issue where any updates to the hooks package causes the page to crash with this error:

`Uncaught DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.`

I'll take a look today and see if I can figure it out. But even with this issue, I still think it'd be worth merging because it's still much quicker to refresh manually for this particular issue than rebuilding the packages on every code change.

Great job figuring this out. Really nice DX improvement 🔥

**Pull request merged**